packages/@sanity/client/src/validators.js

```diff
@@ -2,9 +2,9 @@ const VALID_ASSET_TYPES = ['image', 'file']
 const VALID_INSERT_LOCATIONS = ['before', 'after', 'replace']
 
 exports.dataset = name => {
-  if (!/^[-\w]{1,128}$/.test(name)) {
+  if (!/^[~][a-z0-9][-\w]{2,20}$|^[a-z0-9][-\w]{1,20}$/.test(name)) {
```

Comment:
We're reducing the maximum allowed length of a dataset name by quite a lot. Are we sure that's safe to do without investigating whether users depend on long dataset names or something? :smiley:

Reply:
I asked Espen about this, since I've seen various different validations for dataset. He said max 20 chars, but please confirm again, @rexxars!

Reply:
The API has max 20 characters as the limit, so should be safe!

Reply:
The regex is a bit off though - aliases (should) need between 1 and 26 characters, where the leading character cannot be a dash, eg `^~[a-z0-9]{1}[-\w]{0,25}$`. Datasets are between 1 and 20 characters, eg `^[a-z0-9][-\w]{0,19}`. I believe this catch both cases:

`^(~[a-z0-9]{1}[-\w]{0,25}|[a-z0-9]{1}[-\w]{0,19})$`

Reply:
Thanks for pointing that out. I'll fix that.